trin-state/src/validation/trie.rs

```diff
@@ -49,11 +48,11 @@ pub fn validate_node_trie_proof<'proof, 'path>(
 /// Validate the trie proof associated with the account state.
 pub fn validate_account_state(
     root_hash: Option<B256>,
-    account: &Address,
+    address_hash: &B256,
     proof: &TrieProof,
 ) -> Result<AccountState, StateValidationError> {
-    let path: Vec<u8> = keccak(account)
-        .as_bytes()
+    let path: Vec<u8> = address_hash
+        .as_slice()
```

Comment:
nit: I believe you don't `.as_slice()` and can just call `address_hash.iter()...`.

Also few places below and in the other file.